### PR TITLE
serializers: addition of ``ui`` serializer

### PR DESCRIPTION
**Pull request by drjova**
Signed-off-by: Harris Tzovanakis <me@drjova.com>

### CODE REVIEW

File: inspirehep/modules/records/serializers/json_literature.py [hunk 1]
```diff
@@ -67,27 +56,37 @@ def get_display_fields(record):
...
+
+        ui_metadata = _get_ui_metadata(record)
+
+        if 'authors' in record:
+            record['authors'] = record['authors'][:10]
+        if 'references' in record:
+            del record['references']
+
+        record.update(ui_metadata)
+        result['metadata'] = record
```
Comment (david-caro) [inline inspirehep/modules/records/serializers/json_literature.py]: This should not be needed, though I agree that is more readable like this...

Maybe short-circuiting the if would make it more readable :/, not sure:
```python
def _preprocess_result(result):
    if 'metadata' not in result:
        return result

    record = result['metadata']
    ...
    return result
```

btw. what does it mean that result has no 'metadata'? is that expected? should we throw an exception if not there?

File: inspirehep/modules/records/serializers/json_literature.py [hunk 2]
```diff
@@ -67,27 +56,37 @@ def get_display_fields(record):
...
+
+        # FIXME: Depricated, must be removed once the new UI is released
```
Comment (david-caro) [inline inspirehep/modules/records/serializers/json_literature.py]: s/Depricated/Deprecated/

File: inspirehep/modules/records/serializers/json_literature.py (hunk 1 above)
Comment (drjova) [reply inspirehep/modules/records/serializers/json_literature.py]: Actually ``metadata`` should always be there, is part of the ``preprocess`` https://github.com/inveniosoftware/invenio-records-rest/blob/3d63611e7e85ee5a981ff1faa67d11bc166bdfbe/invenio_records_rest/serializers/base.py#L140-L173

Comment (david-caro) [reply inspirehep/modules/records/serializers/json_literature.py]: Then maybe better throw an exception if it's not there (just removing the check might be enough though, a key error at that point is kind-of understandable too).

Comment (drjova) [reply inspirehep/modules/records/serializers/json_literature.py]: yep, I'll remove it

**Comment by drjova** [comment]
@david-caro updated

**Comment by david-caro** [comment]
:+1:

**Comment by drjova** [comment]
ci merge please

**Pull request merged**